nixos/modules/installer/sd-card/sd-image-riscv64-visionfive.nix

```diff
@@ -0,0 +1,52 @@
+# To build, use:
+# nix-build nixos -I nixos-config=nixos/modules/installer/sd-card/sd-image-riscv64-visionfive.nix -A config.system.build.sdImage
+{ config, lib, pkgs, ... }:
+
+{
+  imports = [
+    ../../profiles/base.nix
+    ./sd-image.nix
+  ];
+
+  hardware.deviceTree.name = "starfive/jh7100-starfive-visionfive-v1.dtb";
+  systemd.services."serial-getty@hvc0".enable = false;
+  environment.systemPackages = with pkgs; [ mtdutils ];
+
+  boot = {
+    consoleLogLevel = lib.mkDefault 7;
+    kernelPackages = pkgs.linuxPackages_visionfive;
+
+    kernelParams = [
+      "console=tty0"
+      "console=ttyS0,115200n8"
+      "earlycon=sbi"
+    ];
+
+    initrd.kernelModules = [
+      "dw-axi-dmac-platform"
+      "dw_mmc-pltfm"
+      "spi-dw-mmio"
+    ];
+
+    loader = {
+      grub.enable = false;
+      generic-extlinux-compatible.enable = true;
+    };
+  };
+
+  sdImage = {
+    imageName = "${config.sdImage.imageBaseName}-${config.system.nixos.label}-${pkgs.stdenv.hostPlatform.system}-visionfive.img";
+
+    # We have to use custom boot firmware since we do not support
+    # StarFive's Fedora MMC partition layout. Thus, we include this in
+    # the image's firmware partition so the user can flash the custom firmware.
+    populateFirmwareCommands = ''
+      cp ${pkgs.firmware-visionfive}/opensbi_u-boot_visionfive.bin firmware/opensbi_u-boot_visionfive.bin
+    '';
```

Comment:
how does one flash this custom firmware? Do I need to follow this [VisionFive Quick Start Guide - Appendix B Updating Firmware and U Boot](https://rvspace.org/en/Product/VisionFive/Technical_Documents/VisionFive_Single_Board_Computer_Quick_Start_Guide#appendix-b-updating-firmware-and-u-boot) or is there an easier way?